### PR TITLE
LOOFAH-5 Connect api to mongo database

### PR DESCRIPTION
**Pull request by Learner5200**
This commit connects the graph api to a mongo database using spring-data-mongodb. For integration tests an embedded database is used for speed and to avoid the need to configure a test database for local integration testing.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -16,28 +16,46 @@
...
+
+		<!--	Test Dependencies	-->
+
 		<dependency>
 			<groupId>org.springframework.boot</groupId>
 			<artifactId>spring-boot-starter-test</artifactId>
 			<scope>test</scope>
 		</dependency>
+		<dependency>
+			<groupId>de.flapdoodle.embed</groupId>
```
Comment (Learner5200) [inline pom.xml:55]: I'm aware that "flapdoodle" is a ridiculous name, but as far as I can tell this is a widely used and trusted library.

File: src/main/java/com/loofah/graph/api/models/Skill.java [hunk 2]
```diff
@@ -36,4 +41,36 @@ public String getDescription() {
     public void setDescription(String description) {
         this.description = description;
     }
+
+    public static SkillBuilder builder() {
+        return new SkillBuilder();
+    }
+
+    public static class SkillBuilder {
```
Comment (Learner5200) [inline src/main/java/com/loofah/graph/api/models/Skill.java:49]: #lifeafterlombok

File: src/main/java/com/loofah/graph/api/config/DatabaseSeeder.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    private SkillsRepository skillsRepository;
+
+    @Autowired
+    public DatabaseSeeder(SkillsRepository skillsRepository) {
+        this.skillsRepository = skillsRepository;
+    }
+
+    @Override
+    public void run(String... args) throws Exception {
+        skillsRepository.save(Skill.builder().withTitle("title1").withDescription( "description1").build());
```
Comment (vijamesDD) [inline src/main/java/com/loofah/graph/api/config/DatabaseSeeder.java]: Does this skill need an Id too like the others?

**Review by vijamesDD — APPROVED** [review]

File: src/main/java/com/loofah/graph/api/config/DatabaseSeeder.java (hunk 3 above)
Comment (Learner5200) [reply src/main/java/com/loofah/graph/api/config/DatabaseSeeder.java]: Ah, that was a test I forgot to remove, to see what would happen if we don't specify an ID (it auto-generates a unique one). I've added the manual ID-setting back in now for consistency.

**Pull request merged**